src/bench/connection.rs

```diff
@@ -172,6 +170,14 @@ impl Connection {
                                 None => warn!("No publish record for Pkid={:?}", index),
                             };
 
+                            // send progress tick over indi_sender
+                            match &self.link.indi_sender.clone(){
+                                Some(v) => {
+                                    v.clone().send(1).await.unwrap();
```

Comment:
Sending progress for every ack doesn't scale for a lot of connections. We should maintain a local counter and periodically send progress.

```rust
loop {
    select! {
        v = eventloop.poll() => { // collect progress }
        _ = periodic_timer => { // send progress }
    }
}
```